csrc/python_frontend/fusion_record.h

```diff
@@ -1406,26 +1485,15 @@ struct OutputRecord : RecordFunctor {
       // With C++17, this statement should be "if constexpr"
       if (std::is_same<OutputType, TensorView>::value) {
         auto tv_output = output->template as<TensorView>();
-
         if (!stride_order_.empty()) {
-          std::vector<int64_t> reverse_perm(stride_order_.size());
-          int64_t duplicate_check = 0;
-          for (const auto i : c10::irange((int64_t)stride_order_.size())) {
-            NVF_CHECK(
-                stride_order_[i] >= 0 &&
-                    stride_order_[i] < (int64_t)reverse_perm.size(),
-                "stride_order elements need to be within [0, stride_order.size())!");
-            reverse_perm[stride_order_[i]] = i;
-            duplicate_check |= 1 << stride_order_[i];
+          size_t rank = stride_order_.size();
```

Comment:
refactored output stride order support to use allocation domain instead, this likely will give us some regression on our support since scheduler doesn't do a good job on that yet.
But I do think this is a step towards the right direction. :crossed_fingers: